crates/shared/src/price_estimation/rate_limited.rs

```diff
@@ -0,0 +1,162 @@
+use crate::{
+    http_client::RateLimiter,
+    price_estimation::{single_estimate, PriceEstimating, PriceEstimationError, Query},
+};
+use futures::stream::{BoxStream, StreamExt};
+use std::{
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+pub trait RateLimitedPriceEstimatorExt {
```

Comment:
I really like how expressive this extension trait makes the code:
```
SomeEstimator::new(...).rate_limited(...)
```

So much style 😎

Reply:
Glad you like it.
I was considering applying this pattern for all the price estimators to see if it would make creating the price estimator hierarchy in the [order book](https://github.com/cowprotocol/services/blob/main/crates/orderbook/src/main.rs#L528-L636) look more reasonable.

Reply:
I prefer standalone functions over extension traits usually. Would only make it a trait if it is used so much that the small ergonomic improvement trumps the more complicated construction. Can still keep it this way here.